packages/pluggableWidgets/rich-text-web/src/components/RichText.tsx

```diff
@@ -1,106 +1,77 @@
-            disableNativeSpellChecker: !props.spellChecker,
-            readOnly: props.readOnly
-        },
         initContent: value,
         dispatchEvent: ({ type, payload }) => {
-            if (type === CKEditorEventAction.change) {
-                const value = payload.editor.getData();
-                if (props.onKeyChange) {
-                    props.onKeyChange();
+            if (type === CKEditorEventAction.key) {
+                if (onKeyPress) {
+                    onKeyPress();
                 }
-                if (props.onKeyPress) {
-                    props.onKeyPress();
-                }
-                if (props?.onValueChange) {
-                    const content = props.sanitizeContent ? sanitizeHtml(value) : value;
-                    props?.onValueChange(content);
+            }
+
+            if (onChange && type === CKEditorEventAction.change) {
+                const valueRaw = payload.editor.getData();
+                const value = editorSettings.sanitizeContent ? sanitizeHtml(valueRaw) : valueRaw;
+
+                localEditorValueRef.current = value;
+
+                if (onChange) {
+                    onChange(value);
                 }
             }
         },
-        subscribeTo: ["change"]
+        subscribeTo: ["change", "key"]
     });
-    const key = useMemo(() => Date.now(), [ckeditorConfig]);
+
     useEffect(() => {
-        const config = { ...props.toolbar };
-        if (plugins?.length) {
-            plugins.forEach((plugin: PluginName) => addPlugin(plugin, config));
-        }
-        if (advancedContentFilter) {
-            config.allowedContent = advancedContentFilter.allowedContent;
-            config.disallowedContent = advancedContentFilter.disallowedContent;
+        if (editor) {
+            if (localEditorValueRef.current !== value) {
+                editor.setData(value);
+                localEditorValueRef.current = value;
+            }
         }
+    }, [editor, localEditorValueRef, value]);
 
-        setCkeditorConfig({
-            ...ckeditorConfig,
-            initContent: value,
-            element,
-            config: {
-                ...ckeditorConfig.config,
-                ...config,
-                readOnly: props.readOnly
-            }
-        });
-    }, [element, value]);
     return (
         <div
-            className={classNames("widget-rich-text", `${readOnly ? `editor-${readOnlyStyle}` : ""}`)}
-            style={{ width, height }}
+            className={classNames("widget-rich-text", readOnlyStyle && `editor-${readOnlyStyle}`)}
+            style={{ width: editorSettings.config?.width, height: editorSettings.config?.height }}
         >
-            <div ref={setElement} id={props.name} />
-            <MainEditor key={key} config={ckeditorConfig} />
+            <div id={name} ref={setElement} style={status !== "ready" ? { visibility: "hidden" } : undefined} />
```

Comment:
why we need to check status of ckeditor for visibility? shouldn't it be covered in the library itself?

Reply:
I took this line from [CKEditor docs example](https://ckeditor.com/docs/ckeditor4/latest/guide/dev_react_v2.html#dispatchevent-explained). In my view this is okay to control this on our side (library don't need to be involved). This trick with visibility just one more way to hide ugly stuff from end user. I think we should keep it.